src/Engine/GameAction.java

```diff
@@ -165,6 +166,12 @@ public String toString()
       return String.format("[Attack %s with %s after moving to %s]",
           defender.toStringWithLocation(), attacker.toStringWithLocation(), moveCoord );
     }
+
+    @Override
+    public UnitActionType getUnitActionType()
```

Comment:
Having both `getType` and `getActionType` feels duplicative.

Reply:
I agree, but I dunno how to fix it.